connect/runtime/src/main/java/org/apache/kafka/connect/runtime/errors/ProcessingContext.java

```diff
@@ -0,0 +1,199 @@
+     * @param record the source record
+     */
+    public void sourceRecord(SourceRecord record) {
+        this.sourceRecord = record;
+        reset();
+    }
+
+    /**
+     * Set the stage in the connector pipeline which is currently executing.
+     *
+     * @param position the stage
+     */
+    public void position(Stage position) {
+        this.position = position;
+    }
+
+    /**
+     * @return the stage in the connector pipeline which is currently executing.
+     */
+    public Stage stage() {
+        return position;
+    }
+
+    /**
+     * @return the class which is going to execute the current operation.
+     */
+    public Class<?> executingClass() {
+        return klass;
+    }
+
+    /**
+     * @param klass set the class which is currently executing.
+     */
+    public void executingClass(Class<?> klass) {
+        this.klass = klass;
+    }
+
+    /**
+     * A helper method to set both the stage and the class.
+     *
+     * @param stage the stage
+     * @param klass the class which will execute the operation in this stage.
+     */
+    public void setCurrentContext(Stage stage, Class<?> klass) {
+        position(stage);
+        executingClass(klass);
+    }
+
+    /**
+     * Report errors. Should be called only if an error was encountered while executing the operation.
+     */
+    public void report() {
+        for (ErrorReporter reporter: reporters) {
+            reporter.report(this);
+        }
+    }
+
+    @Override
+    public String toString() {
+        return "ProcessingContext{" +
+                "position=" + position +
+                ", class=" + klass +
+                ", consumedMessage=" + consumedMessage +
+                ", sourceRecord=" + sourceRecord +
+                ", attempt=" + attempt +
+                ", reporters=" + reporters +
+                '}';
+    }
+
+    /**
+     * @param attempt the number of attempts made to execute the current operation.
+     */
+    public void attempt(int attempt) {
+        this.attempt = attempt;
+    }
+
+    /**
+     * @return the number of attempts made to execute the current operation.
+     */
+    public int attempt() {
+        return attempt;
+    }
+
+    /**
+     * @return the error (if any) which was encountered while processing the current stage.
+     */
+    public Throwable error() {
+        return error;
+    }
+
+    /**
+     * The error (if any) which was encountered while processing the current stage.
+     *
+     * @param error the error
+     */
+    public void error(Throwable error) {
+        this.error = error;
+    }
+
+    /**
+     * @return true, if the last operation encountered an error; false otherwise
+     */
+    public boolean failed() {
+        return error() != null;
+    }
+
+    /**
+     * Set the error reporters for this connector.
+     *
+     * @param reporters the error reporters (should not be null).
+     */
+    public void setReporters(Collection<ErrorReporter> reporters) {
```

Comment:
Doesn't follow AK setter convention. Any reason to maybe set this in an `initialize` method? Seems like it should be called up front, rather than repeatedly. Anything else that should be set only once? Removing unnecessary setters might help with the usability.